qiskit/quantum_info/analysis/sampled_expval.py

```diff
@@ -0,0 +1,89 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Routines for computing expectation values from sampled distributions"""
+import numpy as np
+
+from qiskit.exceptions import QiskitError
+from qiskit.result import Counts, QuasiDistribution, ProbDistribution
+from qiskit.quantum_info import Pauli, SparsePauliOp
+from qiskit.opflow import PauliOp, PauliSumOp
```

Comment:
This smells a bit wrong - `quantum_info` is the lowest package in the Qiskit stack (give or take), so if this module depends on `opflow` and `result`, this code probably needs to be somewhere else.  I get that the naming is unfortunate for that, but from a code organisation perspective, it's probably where your circular imports are coming in.